projects/astc-encoder/astc_load_decompress_image_fuzzer.cc

```diff
@@ -0,0 +1,127 @@
+// Copyright 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "astcenc_internal.h"
+#include "astcenccli_internal.h"
+#include <fuzzer/FuzzedDataProvider.h>
+
+unsigned int unpack_bytes(uint8_t a, uint8_t b, uint8_t c) {
+  return ((unsigned int)a) + ((unsigned int)b << 8) + ((unsigned int)c << 16);
+}
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+
+  if (size < 20)
+    return 0;
+
+  astc_compressed_image image_comp;
+  astcenc_error error;
+  astcenc_profile profile;
+  int out_bitness;
+  FuzzedDataProvider stream(data, size);
+
+  const astcenc_profile profiles[] = {ASTCENC_PRF_LDR, ASTCENC_PRF_LDR_SRGB,
+                                      ASTCENC_PRF_HDR,
+                                      ASTCENC_PRF_HDR_RGB_LDR_A};
+
+  int profile_type = stream.ConsumeIntegralInRange<int>(0, 3);
+  profile = profiles[profile_type];
+  out_bitness = 8 << (profile_type / 2);
+
+  // Avoid dividing by zero
```

Comment:
I checked over the FuzzedDataProvider documentation, and fortunately it has a `ConsumeIntegralInRange` method where you can set the minimum to be 1.  You can use the [`std::numeric_limits`](https://en.cppreference.com/w/cpp/types/numeric_limits) class to get the largest possible `uint8_t` to use as the upper bound.  This way, you can consume a `uint8_t` without needing to check if it's zero.